src/components/learner-credit-management/AssignmentTableRemind.jsx

```diff
@@ -5,17 +5,35 @@ import { Mail } from '@edx/paragon/icons';
 import useRemindContentAssignments from './data/hooks/useRemindContentAssignments';
 import RemindAssignmentModal from './RemindAssignmentModal';
 
-const AssignmentTableRemindAction = ({ selectedFlatRows }) => {
+const calculateTotalToRemind = ({
+  assignmentUuids,
+  isEntireTableSelected,
+  tableItemCount,
+}) => {
+  if (isEntireTableSelected) {
+    return tableItemCount;
+  }
+  return assignmentUuids.length;
+};
+
+const AssignmentTableRemindAction = ({ selectedFlatRows, isEntireTableSelected, tableInstance }) => {
   const assignmentUuids = selectedFlatRows.filter(row => row.original.learnerState === 'waiting').map(({ id }) => id);
   const assignmentConfigurationUuid = selectedFlatRows[0].original.assignmentConfiguration;
-  const selectedRemindableRows = selectedFlatRows.filter(row => row.original.learnerState === 'waiting').length;
   const {
     remindButtonState,
     remindContentAssignments,
     close,
     isOpen,
     open,
-  } = useRemindContentAssignments(assignmentConfigurationUuid, assignmentUuids);
+  } = useRemindContentAssignments(assignmentConfigurationUuid, assignmentUuids, isEntireTableSelected);
+
+  const tableItemCount = tableInstance.itemCount;
+  const selectedRemindableRows = calculateTotalToRemind({
```

Comment:
nit: if this is returning list of assignment uuids now instead of list of full assignment records, should the variable be renamed to something like `totalSelectedRemindableRows` or `selectedRemindableRowsCount`?